apps/camtool/camtool.cpp

```diff
@@ -0,0 +1,225 @@
+
+	std::printf("file: %s\n", cam.GetFilename().c_str());
+	std::printf("file size: %u bytes\n", header.size);
+	std::printf("movement speed: %u\n", header.movementSpeed);
+	std::printf("point count: %u\n", header.pointCount);
+	std::printf("\n");
+
+	return EXIT_SUCCESS;
+}
+
+int PrintPoints(const openblack::cam::CAMFile& cam) noexcept
+{
+	const auto& points = cam.GetPoints();
+
+	std::printf("file: %s\n", cam.GetFilename().c_str());
+	std::printf("%u points {position, heading}\n", cam.GetHeader().pointCount);
+	for (uint32_t i = 0; const auto& point : points)
+	{
+		std::printf("[%u]: {(%f, %f, %f), (%f, %f, %f)}\n", i++, point.position[0], point.position[1], point.heading[2],
+		            point.heading[0], point.heading[1], point.heading[2]);
+	}
+
+	return EXIT_SUCCESS;
+}
+
+int WriteFile(const Arguments::Write& args) noexcept
+{
+	openblack::cam::CAMFile cam {};
+
+	cam.SetMovementSpeed(args.movementSpeed);
+
+	if (args.pointStrings.empty())
+	{
+		return EXIT_FAILURE;
+	}
+	std::vector<openblack::cam::CAMPoint> points;
+	points.reserve(args.pointStrings.size());
+	const auto regex = std::regex(R"( *(-?\d+\.\d+) ?(-?\d+\.\d+) ?(-?\d+\.\d+) ?(-?\d+\.\d+) ?(-?\d+\.\d+) ?(-?\d+\.\d+))");
+	for (const auto& str : args.pointStrings)
+	{
+		std::smatch match;
+		std::regex_match(str, match, regex);
+		if (match.size() != 7)
+		{
+			return EXIT_FAILURE;
+		}
+		auto& point = points.emplace_back();
+		point.position[0] = std::stof(match[1]);
+		point.position[1] = std::stof(match[2]);
+		point.position[2] = std::stof(match[3]);
+		point.heading[0] = std::stof(match[4]);
+		point.heading[1] = std::stof(match[5]);
+		point.heading[2] = std::stof(match[6]);
+	}
+	cam.SetPoints(std::move(points));
+
+	cam.Write(args.outFilename);
+
+	return EXIT_SUCCESS;
+}
+
+bool parseOptions(int argc, char** argv, Arguments& args, int& returnCode) noexcept
+{
+	cxxopts::Options options("camtool", "Inspect and extract files from LionHead camera files.");
+
+	try
+	{
+		options.add_options()                                            //
+		    ("h,help", "Display this help message.")                     //
+		    ("subcommand", "Subcommand.", cxxopts::value<std::string>()) //
+		    ;
+		options.positional_help("[read|write] [OPTION...]");
+		options.add_options("read")                                                                           //
+		    ("H,header", "Print Header Contents.", cxxopts::value<std::vector<std::filesystem::path>>())      //
+		    ("p,print-points", "Print List of Points.", cxxopts::value<std::vector<std::filesystem::path>>()) //
+		    ;
+		options.add_options("write")                                                          //
+		    ("o,output", "Output file (required).", cxxopts::value<std::filesystem::path>())  //
+		    ("s,movement-speed", "Value for the movement speed.", cxxopts::value<uint32_t>()) //
+		    ("points", "Point in format \"p1 p2 p3 h1 h2 h3\" array (comma-separated).",      //
```

Comment:
don't need to say comma-separated because cxxopts will do the same with `--points "1.0 2.0 3.0 4.0 5.0 6.0, 11.0 12.0 13.0 14.0 15.0 16.0"` and `--points "1.0 2.0 3.0 4.0 5.0 6.0" --points "11.0 12.0 13.0 14.0 15.0 16.0"`